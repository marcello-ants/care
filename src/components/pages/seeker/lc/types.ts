import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
