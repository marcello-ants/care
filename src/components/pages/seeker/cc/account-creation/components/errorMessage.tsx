import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
