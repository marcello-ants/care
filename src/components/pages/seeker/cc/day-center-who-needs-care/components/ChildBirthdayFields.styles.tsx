// External Dependencies
import { makeStyles } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

const useStyles = makeStyles({
  childContainer: {
    padding: theme.spacing(0),
  },
  childContainerNth: {
    marginTop: theme.spacing(4),
    position: 'relative',
  },
  optionsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  fullWidth: {
    width: '100%',
  },
  error: {
    color: theme.palette.care.red[800],
    paddingLeft: theme.spacing(1),
  },
  selectContainer: {
    marginTop: theme.spacing(1),
  },
  monthSelect: {
    marginRight: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  yearSelect: {
    marginLeft: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  removeChildButton: {
    position: 'absolute',
    top: '0',
    right: '0',
    width: 'auto',
    height: 'auto',
    lineHeight: 'unset',
    minWidth: 'auto',
    padding: theme.spacing(0),
    color: theme.palette.care.blue[700],
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
});

export default useStyles;
