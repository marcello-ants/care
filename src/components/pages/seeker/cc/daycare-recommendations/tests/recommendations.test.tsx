// External Dependencies
import React from 'react';
import { cloneDeep } from 'lodash-es';
import { render, screen, waitFor } from '@testing-library/react';
import { NextRouter, useRouter } from 'next/router';
import userEvent from '@testing-library/user-event';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

// Internal Dependencies
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import AuthService from '@/lib/AuthService';
import { ChildCareLeadsSubmissionMethod, DayCareTimeOfDay } from '@/__generated__/globalTypes';
import { DaycareProviderProfile, DayCareFrequencyOptions, CareDayTime } from '@/types/seekerCC';
import { AppState } from '@/types/app';
import { initialAppState } from '@/state';
import {
  POST_AUTO_ACCEPT_CHILD_CARE_LEAD,
  POST_CHILD_CARE_LEAD,
  UPDATE_SEEKER_ATTRIBUTE,
} from '@/components/request/GQL';
import { AppStateProvider } from '@/components/AppState';
import useTealiumTracking from '@/components/hooks/useTealiumTracking';
import { mapMonthToDate } from '@/utilities/account-creation-utils';
import logger from '@/lib/clientLogger';
import Recommendations from '../recommendations';

jest.mock('@/lib/AuthService');
const AuthServiceMock = AuthService as jest.Mock;

const getStoreMock = jest.fn();
AuthServiceMock.mockImplementation(() => {
  return {
    getStore: getStoreMock,
  };
});

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

jest.mock('next/config', () => () => {
  return {
    publicRuntimeConfig: {
      CZEN_GENERAL: '/',
    },
  };
});

jest.mock('@/components/hooks/useTealiumTracking');

let mockRouter: Pick<NextRouter, 'push' | 'asPath' | 'pathname'>;
const TEST_EMAIL_MOCK = '123@care.com';
const START_MONTH_MOCK = 'JANUARY';
const START_DATE_MOCK = mapMonthToDate(START_MONTH_MOCK);
const PROVIDER_ID_MOCK = '70bed9b5-e5b8-4fd3-a205-44c69758fb7e';
const TRACKING_ID_MOCK = '44c69758-4fd3-e5b8-a205-70bed9b5a72f';
const ZIPCODE_MOCK = '12345';
const ADDITIONAL_INFORMATION_MOCK = '';
const MEMBER_ID_MOCK = '11111';
const SEEKER_UUID_MOCK = '15602b3a-a77b-4983-9251-23c07d1c698f';
const FIRSTNAME_MOCK = 'dev';
const LASTNAME_MOCK = 'test';
const PHONENUMBER_MOCK = '6191234567';
const CHILDREN_DOB_MOCK = '2015-01-15';
const TIME_OF_DAY_MOCK = DayCareTimeOfDay.MORNING;
const SOURCE_MOCK = 'ENROLLMENT';
const ATTENDING_DAYS_MOCK = {
  monday: true,
  tuesday: true,
  wednesday: true,
  thursday: true,
  friday: true,
  saturday: false,
  sunday: false,
};

const postChildcareLeadMock = {
  request: {
    query: POST_CHILD_CARE_LEAD,
    variables: {
      childCareLeadInput: {
        submissionMethod: ChildCareLeadsSubmissionMethod.MANUAL,
        providerIds: [PROVIDER_ID_MOCK],
        trackingId: TRACKING_ID_MOCK,
        zipcode: ZIPCODE_MOCK,
        source: SOURCE_MOCK,
        childCareExpectations: ADDITIONAL_INFORMATION_MOCK,
        startDate: START_DATE_MOCK,
        childrenDatesOfBirth: [CHILDREN_DOB_MOCK],
        timeOfDay: TIME_OF_DAY_MOCK,
        attendingDays: ATTENDING_DAYS_MOCK,
        seekerInfo: {
          id: SEEKER_UUID_MOCK,
          email: TEST_EMAIL_MOCK,
          firstName: FIRSTNAME_MOCK,
          lastName: LASTNAME_MOCK,
          phoneNumber: PHONENUMBER_MOCK,
        },
      },
    },
  },
  result: {
    data: {
      childCareLeadCreate: {
        batchId: '1234567890',
      },
    },
  },
};

const postAutoAcceptChildcareLeadMock = {
  request: {
    query: POST_AUTO_ACCEPT_CHILD_CARE_LEAD,
    variables: {
      childCareLeadInput: {
        submissionMethod: ChildCareLeadsSubmissionMethod.MANUAL,
        providerIds: [PROVIDER_ID_MOCK],
        trackingId: TRACKING_ID_MOCK,
        zipcode: ZIPCODE_MOCK,
        source: SOURCE_MOCK,
        childCareExpectations: ADDITIONAL_INFORMATION_MOCK,
        startDate: START_DATE_MOCK,
        childrenDatesOfBirth: [CHILDREN_DOB_MOCK],
        timeOfDay: TIME_OF_DAY_MOCK,
        attendingDays: ATTENDING_DAYS_MOCK,
        seekerInfo: {
          id: SEEKER_UUID_MOCK,
          email: TEST_EMAIL_MOCK,
          firstName: FIRSTNAME_MOCK,
          lastName: LASTNAME_MOCK,
          phoneNumber: PHONENUMBER_MOCK,
        },
      },
    },
  },
  result: {
    data: {
      childCareLeadCreate: {
        batchId: '1234567890',
      },
    },
  },
};

const postChildcareLeadErrorMock = {
  request: {
    query: POST_CHILD_CARE_LEAD,
    variables: {
      childCareLeadInput: {
        submissionMethod: ChildCareLeadsSubmissionMethod.MANUAL,
        providerIds: [PROVIDER_ID_MOCK],
        trackingId: TRACKING_ID_MOCK,
        zipcode: ZIPCODE_MOCK,
        source: SOURCE_MOCK,
        childCareExpectations: ADDITIONAL_INFORMATION_MOCK,
        startDate: START_DATE_MOCK,
        childrenDatesOfBirth: [CHILDREN_DOB_MOCK],
        timeOfDay: TIME_OF_DAY_MOCK,
        attendingDays: ATTENDING_DAYS_MOCK,
        seekerInfo: {
          id: SEEKER_UUID_MOCK,
          email: TEST_EMAIL_MOCK,
          firstName: FIRSTNAME_MOCK,
          lastName: LASTNAME_MOCK,
          phoneNumber: PHONENUMBER_MOCK,
        },
      },
    },
  },
  result: {
    data: null,
  },
  error: new Error('An error occurred'),
};

const updateSeekerAttributeMock = {
  request: {
    query: UPDATE_SEEKER_ATTRIBUTE,
    variables: {
      input: { attributes: { daycareInterest: true } },
    },
  },
  result: {
    data: {
      seekerUpdate: {
        success: true,
      },
    },
  },
};

function renderPage(initialState: AppState, mocks: MockedResponse[] = [], ldFlags?: FeatureFlags) {
  const pathname = '/seeker/dc/recommendations';
  mockRouter = {
    push: jest.fn(),
    asPath: pathname,
    pathname,
  };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);
  getStoreMock.mockReturnValue({
    profile: { email: TEST_EMAIL_MOCK, sub: `mem:${SEEKER_UUID_MOCK}` },
  });

  const flags = {
    'cc-enrollment-mfe-daycare-request-tours-copy': {
      reason: { kind: 'FALLTHROUGH' },
      value: 'control',
      variationIndex: 0,
    },
    'enrollment-mfe-daycare-auto-accept-leads-backend': {
      value: false,
      reason: { kind: 'FALLTHROUGH' },
    },
    'growth-enrollment-mfe-daycare-leads-redirection': {
      value: false,
      reason: { kind: 'FALLTHROUGH' },
    },
  };

  return render(
    <AppStateProvider initialStateOverride={initialState}>
      <ThemeProvider theme={theme}>
        <FeatureFlagsProvider flags={ldFlags || flags}>
          <MockedProvider mocks={[...mocks, updateSeekerAttributeMock]} addTypename>
            <Recommendations />
          </MockedProvider>
        </FeatureFlagsProvider>
      </ThemeProvider>
    </AppStateProvider>
  );
}

const appState = cloneDeep(initialAppState);
const recommendations: DaycareProviderProfile[] = [
  {
    __typename: 'Provider',
    id: '70bed9b5-e5b8-4fd3-a205-44c69758fb7e',
    description:
      'The Transportation Childrens Center, a non-profit child care facility, in downtown Boston, provides full day Infant, Toddler, Preschool and Pre-Kindergarten Programs year round. Open to the general public, the center is available to children 3 months through age five. The center is open from 8:00 a.m. to 6:00 p.m. with both part time and full time programs available. Every effort is made to keep tuition rates as low as possible while maintaining high quality services. The established and long-term professional staff hold degrees in early education and have extensive experience working with young children. The facility provides an outdoor roof-top play space, drop-off parking and is accessible by the MBTA.\r\nEstablished since 1986 the center is NAEYC Accredited. The beautifully designed and well-equipped center provides a stimulating and nurturing environment which will enhance the childrens self-esteem and future success in school. TCC has implemented the Creative Curriculum philosophy, a hands-on approach to learning and discovering where activities are planned to assure optimum growth and development of the whole child. Slots are filled on a first come, first served basis according to status on the wait list. Families are encouraged to visit the center and schedule a tour.',
    name: 'CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY',
    logo: null,
    images: null,
    address: null,
    reviews: null,
    avgReviewRating: 3,
    license: {
      __typename: 'License',
      name: 'Child Care',
      verifiedDate: '2021/1/1',
      certified: true,
      externalUrl: 'http://childcaresearch.ohio.gov/',
      administrativeArea: 'Ohio',
    },
    selected: false,
    hasCoordinates: true,
    centerType: null,
  },
];

const initialState: AppState = {
  ...appState,
  seekerCC: {
    ...appState.seekerCC,
    dayCare: {
      ...appState.seekerCC.dayCare,
      recommendations,
    },
  },
};

describe('Seeker - Day Care - Recommendations', () => {
  const originalScrollTo = window.scrollTo;
  const originalLocation = window.location;
  const windowScrollToMock = jest.fn();
  const windowLocationAssignMock = jest.fn();
  const loggerErrorMock = jest.fn();

  beforeAll(() => {
    // @ts-ignore
    delete window.scrollTo;
    // @ts-ignore
    window.scrollTo = windowScrollToMock;
    // @ts-ignore
    delete window.location;
    // @ts-ignore
    window.location = {
      assign: windowLocationAssignMock,
    };
    logger.error = loggerErrorMock;
  });

  afterAll(() => {
    window.scrollTo = originalScrollTo;
    window.location = originalLocation;
  });

  it('matches snapshot', async () => {
    const view = renderPage(initialState);

    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: '1. CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY' })
      ).toBeInTheDocument();
    });

    expect(useTealiumTracking).toHaveBeenCalledWith(['/us-marketplace/daycare/leads-viewed/'], {
      tealium_event: 'DAYCARE_LEADS_VIEWED',
      intent: 'WITHIN_A_WEEK',
      leadCountNational: 0,
      leadCountSMB: 0,
    });
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('opens profile details when selecting a daycare result XXXXXXXX', async () => {
    renderPage(initialState);

    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: '1. CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY' })
      ).toBeInTheDocument();
    });

    userEvent.click(
      screen.getByRole('heading', { name: '1. CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY' })
    );

    await waitFor(() => {
      expect(screen.getByText('Licensing Information')).toBeInTheDocument();
    });
  });

  it('clicking Request Info without selecting results shows an error alert', async () => {
    renderPage(initialState);

    userEvent.click(screen.getByRole('button', { name: 'Request info' }));

    await waitFor(() => {
      expect(screen.getByText('Please select at least 1 daycare to proceed')).toBeInTheDocument();
    });
  });

  it('clicking Request Info with at least one daycare selected redirects to next step', async () => {
    const state = {
      ...initialAppState,
      flow: {
        ...initialAppState.flow,
        memberId: MEMBER_ID_MOCK,
      },
      seeker: {
        ...initialAppState.seeker,
        zipcode: ZIPCODE_MOCK,
      },
      seekerCC: {
        ...initialAppState.seekerCC,
        phoneNumber: PHONENUMBER_MOCK,
        firstName: FIRSTNAME_MOCK,
        lastName: LASTNAME_MOCK,
        dayCare: {
          ...initialAppState.seekerCC.dayCare,
          recommendations,
          recommendationsTrackingId: TRACKING_ID_MOCK,
          startMonth: START_MONTH_MOCK,
          additionalInformation: ADDITIONAL_INFORMATION_MOCK,
          dayTime: CareDayTime.MORNING,
          careFrequency: {
            ...initialAppState.seekerCC.dayCare.careFrequency,
            careType: DayCareFrequencyOptions.FULL_TIME,
          },
          childrenDateOfBirth: [{ year: '2015', month: '01' }],
        },
      },
    };
    renderPage(state, [postChildcareLeadMock]);

    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: '1. CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY' })
      ).toBeInTheDocument();
    });

    userEvent.click(screen.getByRole('checkbox', { name: '' }));
    userEvent.click(screen.getByRole('button', { name: 'Request info' }));

    expect(screen.queryByRole('button', { name: 'Next' })).not.toBeInTheDocument();

    await waitFor(() => {
      expect(mockRouter.push).toHaveBeenCalled();
    });
  });

  it('clicking Request Info does not redirect to next step if mutation fails', async () => {
    const state = {
      ...initialAppState,
      flow: {
        ...initialAppState.flow,
        memberId: MEMBER_ID_MOCK,
      },
      seeker: {
        ...initialAppState.seeker,
        zipcode: ZIPCODE_MOCK,
      },
      seekerCC: {
        ...initialAppState.seekerCC,
        phoneNumber: PHONENUMBER_MOCK,
        firstName: FIRSTNAME_MOCK,
        lastName: LASTNAME_MOCK,
        dayCare: {
          ...initialAppState.seekerCC.dayCare,
          recommendations,
          recommendationsTrackingId: TRACKING_ID_MOCK,
          startMonth: START_MONTH_MOCK,
          additionalInformation: ADDITIONAL_INFORMATION_MOCK,
          dayTime: CareDayTime.MORNING,
          careFrequency: {
            ...initialAppState.seekerCC.dayCare.careFrequency,
            careType: DayCareFrequencyOptions.FULL_TIME,
          },
          childrenDateOfBirth: [{ year: '2015', month: '01' }],
        },
      },
    };
    renderPage(state, [postChildcareLeadErrorMock]);

    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: '1. CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY' })
      ).toBeInTheDocument();
    });

    userEvent.click(screen.getByRole('checkbox', { name: '' }));
    userEvent.click(screen.getByRole('button', { name: 'Request info' }));

    await waitFor(() => {
      expect(loggerErrorMock).toHaveBeenCalled();
    });
    await waitFor(() => {
      expect(mockRouter.push).not.toHaveBeenCalled();
    });
  });

  it('clicking Next redirects to next step', async () => {
    const state: AppState = {
      ...initialState,
      seekerCC: {
        ...initialState.seekerCC,
        dayCare: {
          ...initialState.seekerCC.dayCare,
          submissionCompleted: true,
        },
      },
    };
    renderPage(state);

    userEvent.click(screen.getByRole('button', { name: 'Next' }));

    await waitFor(() => {
      expect(mockRouter.push).toHaveBeenCalled();
    });
  });

  it('clicking checkbox toggles them on and off', () => {
    renderPage(initialState);

    expect(screen.getByRole('checkbox', { name: '' })).not.toBeChecked();
    userEvent.click(screen.getByRole('checkbox', { name: '' }));
    expect(screen.getByRole('checkbox', { name: '' })).toBeChecked();
    userEvent.click(screen.getByRole('checkbox', { name: '' }));
    expect(screen.getByRole('checkbox', { name: '' })).not.toBeChecked();
  });

  it('clicking checkbox has no effect when leads have been submitted', () => {
    const state: AppState = {
      ...initialState,
      seekerCC: {
        ...initialState.seekerCC,
        dayCare: {
          ...initialState.seekerCC.dayCare,
          recommendations: initialState.seekerCC.dayCare.recommendations.map((next) => ({
            ...next,
            selected: false,
          })),
          submissionCompleted: true,
        },
      },
    };
    renderPage(state);

    expect(screen.getByRole('checkbox', { name: '' })).toBeDisabled();
    userEvent.click(screen.getByRole('checkbox', { name: '' }), undefined, {
      skipPointerEventsCheck: true,
    });
    expect(screen.getByRole('checkbox', { name: '' })).not.toBeChecked();
  });

  it('daycare selections is maintained', () => {
    const state: AppState = {
      ...initialState,
      seekerCC: {
        ...initialState.seekerCC,
        dayCare: {
          ...initialState.seekerCC.dayCare,
          recommendations: initialState.seekerCC.dayCare.recommendations.map((next) => ({
            ...next,
            selected: true,
          })),
          submissionCompleted: true,
        },
      },
    };
    renderPage(state);

    expect(screen.getByRole('checkbox', { name: '' })).toBeChecked();
    expect(screen.queryByRole('button', { name: 'Request info' })).not.toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Next' })).toBeInTheDocument();
  });

  it('redirects to nth day and logs an error when required data is missing', async () => {
    const selectedRecommendations = recommendations.map((el) => ({
      ...el,
      selected: true,
    }));
    const state = {
      ...initialAppState,
      flow: {
        ...initialAppState.flow,
        memberId: MEMBER_ID_MOCK,
      },
      seekerCC: {
        ...initialAppState.seekerCC,
        dayCare: {
          ...initialAppState.seekerCC.dayCare,
          recommendations: selectedRecommendations,
          recommendationsTrackingId: TRACKING_ID_MOCK,
        },
      },
    };
    const flags = {
      'enrollment-mfe-daycare-auto-accept-leads-backend': {
        value: true,
        reason: { kind: 'FALLTHROUGH' },
      },
      'growth-enrollment-mfe-daycare-leads-redirection': {
        value: true,
        reason: { kind: 'FALLTHROUGH' },
      },
    };
    renderPage(state, [postAutoAcceptChildcareLeadMock], flags);

    expect(loggerErrorMock).toHaveBeenCalledWith({
      event: 'childCareDayCareLeadCreationMissingData',
      data: {
        email: 'exists',
        phoneNumber: 'empty',
        firstName: '',
        lastName: '',
        memberId: MEMBER_ID_MOCK,
        seekerUuid: SEEKER_UUID_MOCK,
        zipcode: '',
        recommendationsTrackingId: TRACKING_ID_MOCK,
        childrenDateOfBirth: [],
        recommendations: selectedRecommendations,
        careType: DayCareFrequencyOptions.FULL_TIME,
        distance: initialAppState.seeker.maxDistanceFromSeekerDayCare,
      },
    });
    expect(windowLocationAssignMock).toHaveBeenCalledWith('//app/job/dc');
  });

  it('redirects to nth day and logs an error when required data is missing on Requesting info click', async () => {
    const selectedRecommendations = recommendations.map((el) => ({
      ...el,
      selected: true,
    }));
    const state = {
      ...initialAppState,
      flow: {
        ...initialAppState.flow,
        memberId: MEMBER_ID_MOCK,
      },
      seekerCC: {
        ...initialAppState.seekerCC,
        dayCare: {
          ...initialAppState.seekerCC.dayCare,
          recommendations: selectedRecommendations,
          recommendationsTrackingId: TRACKING_ID_MOCK,
        },
      },
    };
    const flags = {
      'enrollment-mfe-daycare-auto-accept-leads-backend': {
        value: false,
        reason: { kind: 'FALLTHROUGH' },
      },
      'growth-enrollment-mfe-daycare-leads-redirection': {
        value: true,
        reason: { kind: 'FALLTHROUGH' },
      },
    };
    renderPage(state, [postChildcareLeadMock], flags);

    userEvent.click(screen.getByRole('button', { name: 'Request info' }));

    expect(loggerErrorMock).toHaveBeenCalledWith({
      event: 'childCareDayCareLeadCreationMissingData',
      data: {
        email: 'exists',
        phoneNumber: 'empty',
        firstName: '',
        lastName: '',
        memberId: MEMBER_ID_MOCK,
        seekerUuid: SEEKER_UUID_MOCK,
        zipcode: '',
        recommendationsTrackingId: TRACKING_ID_MOCK,
        childrenDateOfBirth: [],
        recommendations: selectedRecommendations,
        careType: DayCareFrequencyOptions.FULL_TIME,
      },
    });
    expect(windowLocationAssignMock).toHaveBeenCalledWith('//app/job/dc');
  });
});
