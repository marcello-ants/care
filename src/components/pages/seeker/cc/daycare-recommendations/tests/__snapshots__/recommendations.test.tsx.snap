// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seeker - Day Care - Recommendations matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <div
          class="makeStyles-root-7"
        >
          <div
            class="makeStyles-content-8 makeStyles-confirmation-15 makeStyles-roundCorners-10"
            style="width: 100%;"
          >
            <div
              class="makeStyles-icon-11"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-97"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 12a10 10 0 1110 10A10 10 0 012 12zm1.32 0A8.68 8.68 0 1012 3.32 8.7 8.7 0 003.32 12z"
                  fill="#00a979"
                />
                <path
                  d="M10.35 15.3a.68.68 0 01-.36-.18l-2.43-2.64a.54.54 0 01.09-.62.45.45 0 01.63 0l2.07 2.29 4.78-5a.45.45 0 01.63 0 .41.41 0 010 .62l-5 5.37a.49.49 0 01-.41.16z"
                  fill="#00a979"
                />
                <path
                  d="M10.35 15.55a.91.91 0 01-.53-.25l-2.48-2.71a.77.77 0 01.13-.9.7.7 0 011 0l1.9 2.11L15 9a.7.7 0 011 0 .68.68 0 01.21.49.67.67 0 01-.21.49l-5 5.37a.77.77 0 01-.65.2zm-2.58-3.21L10.18 15a.54.54 0 00.18.1c.07 0 .1 0 .18-.1l5-5.37a.19.19 0 00.07-.14.18.18 0 00-.06-.13.19.19 0 00-.28 0l-5 5.21L8.1 12a.2.2 0 00-.28 0 .28.28 0 00-.05.34z"
                  fill="#00a979"
                />
              </svg>
            </div>
            <div>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Account created!
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-46"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiBox-root MuiBox-root-60"
        >
          <h2
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h2"
          >
            Get started evaluating these daycares!
          </h2>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
        >
          <div
            class="MuiBox-root MuiBox-root-61"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
        >
          <div
            class="makeStyles-cardContainer-1"
          >
            <div
              class="MuiGrid-root makeStyles-daycare-62 MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-74"
                    >
                      <div
                        class="makeStyles-daycareInfo-66"
                      >
                        <div
                          aria-label="License Verified"
                          class="makeStyles-root-80 makeStyles-badge-78"
                          role="tooltip"
                          style="background-color: rgb(255, 243, 194);"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root makeStyles-root-98"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10.27 16.83a1 1 0 01-.7-.3l-4.3-4.3a1 1 0 011.41-1.41l3.59 3.59 6.8-6.79A1 1 0 0118.48 9L11 16.53a1 1 0 01-.73.3z"
                              fill="#172c42"
                            />
                          </svg>
                          <p
                            class="MuiTypography-root makeStyles-badge-26 MuiTypography-body1"
                          >
                            License Verified
                          </p>
                        </div>
                        <div
                          aria-label="3.0"
                          class="makeStyles-root-80 makeStyles-badge-78"
                          role="tooltip"
                          style="background-color: rgb(255, 243, 194);"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root makeStyles-root-99 makeStyles-ratingIcon-77"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.729 7.218l.477-1.305 1.212-3.318C10.775 1.62 11.325 1 12.098 1c.791 0 1.337.64 1.672 1.64l1.123 3.35.706 2.106 4.017.134 1.777.06c2.299.076 2.848 1.834.991 3.192l-1.432 1.048-1.433 1.048-1.795 1.314.23.794c.198.667.323 1.094.38 1.337l.992 3.39c.398 1.374.038 2.525-1.21 2.525-.477 0-.986-.213-1.527-.598l-4.715-3.348-4.803 3.222c-.535.36-1.036.557-1.5.557-1.274 0-1.626-1.173-1.187-2.554l.54-1.685c.304-.957.467-1.47.541-1.685l.67-2.11-1.762-1.363-2.805-2.17C-.25 9.798.348 8.057 2.646 8.041l1.776-.012 4.02-.028.287-.783z"
                            />
                          </svg>
                          <p
                            class="MuiTypography-root makeStyles-badge-26 MuiTypography-body1"
                          >
                            3.0
                          </p>
                        </div>
                        <div
                          aria-label="Daycare center"
                          class="makeStyles-root-80 makeStyles-badge-78"
                          role="tooltip"
                          style="background-color: rgb(231, 240, 251);"
                        >
                          <p
                            class="MuiTypography-root makeStyles-badge-26 MuiTypography-body1"
                          >
                            Daycare center
                          </p>
                        </div>
                      </div>
                      <div
                        class="makeStyles-daycareAddressContainer-67"
                      >
                        <div
                          class="makeStyles-daycareName-72"
                        >
                          <h4
                            class="MuiTypography-root MuiTypography-h4"
                          >
                            1. CARE AFTER SCHOOL - COLONIAL HILLS ELEMENTARY
                          </h4>
                        </div>
                      </div>
                    </div>
                    <img
                      alt="Daycare profile"
                      class="makeStyles-iconImage-84 makeStyles-iconImageMobile-85"
                      src="/app/enrollment/recommendationCard/icons/DTMW_DC_generic01.png"
                    />
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-daycareCheckBox-73 MuiGrid-item MuiGrid-grid-xs-2"
                  >
                    <label
                      class="MuiFormControlLabel-root makeStyles-root-86"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-87 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            class="PrivateSwitchBase-input-90"
                            data-indeterminate="false"
                            name="70bed9b5-e5b8-4fd3-a205-44c69758fb7e-selected"
                            type="checkbox"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root makeStyles-root-100"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                              fill="#fff"
                            />
                            <path
                              d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                              fill="#23405f"
                            />
                          </svg>
                        </span>
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      />
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
    >
      <div
        class="MuiBox-root MuiBox-root-92"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-nextButton-2 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Request info
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
